azurelinuxagent/common/cgroups/cgroups.py

```diff
@@ -748,7 +617,7 @@ def set_cpu_limit(self, limit=None):
         if not CGroups.enabled():
             return
 
-        if limit is None:
+        if not limit:
```

Comment:
why the change to "not"? thanks
btw given this test we should probably remove the default value for limit (None)

Reply:
> why the change to "not"? thanks
Making it pythonic.

>btw given this test we should probably remove the default value for limit (None)
`None` could still be passed from the caller function. I'll check the code, and try to see if theres a test which covers it.

Reply:
not that it matters, but I think the best practice is the comparison against None (i.e. the way the code was written) -- no need to change, i was just curious.

for the default value -- the design is kinda weird.... set_cpu_limit with no arguments is a no-op... why allow that at all?

Reply:
>for the default value -- the design is kinda weird.... set_cpu_limit with no arguments is a no-op... why allow that at all?
makes sense. Let me take a look at this piece of code, and how it handles the set-limit in the new world order.

Reply:
@narrieta - This is now done. The HandlerConfig is validated at load time. After going through the usecases for tests, it made sense to load and report the issues immediately. Thanks.

Reply:
any updates?

Reply:
This is fixed in the new code -where the defaults also come from top. limits should now always be populated.